### PR TITLE
Add Drupal Console. Address #357

### PR DESCRIPTION
**Pull request by ruebot**
@DiegoPino @whikloj @bryjbrown -- feel free to test :smile:

### CODE REVIEW

**Comment by bryjbrown** [comment]
@ruebot Testing now.

**Comment by whikloj** [comment]
`vagrant up`

**Comment by whikloj** [comment]
ERROR ERROR
```
==> default: Installing Islandora Karaf Components
==> default: Installing Islandora Sync Gateway
==> default: 693 [sshd-SshClient[7b1d7fff]-nio2-thread-1] WARN org.apache.sshd.client.keyverifier.AcceptAllServerKeyVerifier - Server at /0.0.0.0:8101 presented unverified RSA key: 9c:62:61:f6:08:bf:6f:a0:0d:f6:17:ea:af:da:c1:a3
==> default: Adding feature url file:/home/vagrant/Alpaca/karaf/target/classes/features.xml
==> default: Error executing command: Unable to resolve root: missing requirement [root] osgi.identity; osgi.identity=islandora-sync-gateway; type=karaf.feature; version="[0.1.0.SNAPSHOT,0.1.0.SNAPSHOT]"; filter:="(&(osgi.identity=islandora-sync-gateway)(type=karaf.feature)(version>=0.1.0.SNAPSHOT)(version<=0.1.0.SNAPSHOT))" [caused by: Unable to resolve islandora-sync-gateway/0.1.0.SNAPSHOT: missing requirement [islandora-sync-gateway/0.1.0.SNAPSHOT] osgi.identity; osgi.identity=ca.islandora.sync.islandora-sync-gateway; type=osgi.bundle; version="[0.1.0.SNAPSHOT,0.1.0.SNAPSHOT]"; resolution:=mandatory [caused by: Unable to resolve ca.islandora.sync.islandora-sync-gateway/0.1.0.SNAPSHOT: missing requirement [ca.islandora.sync.islandora-sync-gateway/0.1.0.SNAPSHOT] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.apache.camel.component.http)(version>=2.15.1)(!(version>=3.0.0)))"]]
==> default: 
==> default: 
==> default: Installing Islandora Triplestore Indexer
==> default: 452 [sshd-SshClient[7b1d7fff]-nio2-thread-2] WARN org.apache.sshd.client.keyverifier.AcceptAllServerKeyVerifier - Server at /0.0.0.0:8101 presented unverified RSA key: 9c:62:61:f6:08:bf:6f:a0:0d:f6:17:ea:af:da:c1:a3
==> default: Adding feature url file:/home/vagrant/Alpaca/karaf/target/classes/features.xml
==> default: Error executing command: Unable to resolve root: missing requirement [root] osgi.identity; osgi.identity=islandora-indexing-triplestore; type=karaf.feature; version="[0.1.0.SNAPSHOT,0.1.0.SNAPSHOT]"; filter:="(&(osgi.identity=islandora-indexing-triplestore)(type=karaf.feature)(version>=0.1.0.SNAPSHOT)(version<=0.1.0.SNAPSHOT))" [caused by: Unable to resolve islandora-indexing-triplestore/0.1.0.SNAPSHOT: missing requirement [islandora-indexing-triplestore/0.1.0.SNAPSHOT] osgi.identity; osgi.identity=ca.islandora.indexing.islandora-indexing-triplestore; type=osgi.bundle; version="[0.1.0.SNAPSHOT,0.1.0.SNAPSHOT]"; resolution:=mandatory [caused by: Unable to resolve ca.islandora.indexing.islandora-indexing-triplestore/0.1.0.SNAPSHOT: missing requirement [ca.islandora.indexing.islandora-indexing-triplestore/0.1.0.SNAPSHOT] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.fcrepo.camel)(version>=4.4.0)(!(version>=5.0.0)))"]]
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

**Comment by dannylamb** [comment]
@ruebot keep the issues seperate

**Comment by whikloj** [comment]
`vagrant up`

File: install/Vagrantfile [hunk 1]
```diff
@@ -74,13 +61,12 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
...
   config.vm.provision :shell, :path => "./scripts/loris.sh", :args => home_dir
   config.vm.provision :shell, :path => "./scripts/composer.sh", :args => home_dir
+  config.vm.provision :shell, :path => "./scripts/drupal-console.sh", :args => home_dir
   config.vm.provision :shell, :path => "./scripts/drupal.sh", :args => home_dir
   config.vm.provision :shell, :path => "./scripts/fcrepo.sh", :args => home_dir
   config.vm.provision :shell, :path => "./scripts/blazegraph.sh", :args => home_dir
   config.vm.provision :shell, :path => "./scripts/alpaca.sh", :args => home_dir
   config.vm.provision :shell, :path => "./scripts/karaf.sh", :args => home_dir
-  config.vm.provision :shell, :path => "./scripts/islandora-karaf-components.sh", :args => home_dir
-  config.vm.provision :shell, :path => "./scripts/config.sh", :args => home_dir
```
Comment (whikloj) [inline install/Vagrantfile]: We need this to remain or Fedora won't index to the triplestore and the microservices can't resolve UUIDs.

Comment (whikloj) [inline install/Vagrantfile]: And actually, without the islandora karaf modules, nothing pulls in fcrepo-indexing-triplestore service. So we probably need to `repo:add` and `feature:install` that particular service to support the configuration parameters we are changing in `config.sh`

File: install/scripts/post-install.sh [hunk 2]
```diff
@@ -21,12 +21,3 @@ chown -R vagrant:vagrant "$HOME_DIR"
 # Cycle tomcat
 cd /var/lib/tomcat7
 service tomcat7 restart
-
-# Cycle karaf and watch the maven bundles
-service karaf-service restart
```
Comment (whikloj) [inline install/scripts/post-install.sh]: Along with keeping the `fcrepo-indexing-triplestore` service above, we might want to re-add this restart.

**Comment by whikloj** [comment]
@ruebot Okay but to be clear none of the microservices will work if they can't resolve the UUID -> fedora 4 path

**Comment by DiegoPino** [comment]
@whikloj i agree. At least for the next sprint, having UUID solving is needed. Almost impossible to make drupal + microservices work without (well almost, because if inside a TX and doing POST against root it will work thanks to your last pull in crayfish)

**Comment by dannylamb** [comment]
I'd keep karaf around just to deploy the triple store indexer.

**Comment by dannylamb** [comment]
Don't install sync gateway, just the triplestore indexer.

**Comment by dannylamb** [comment]
And by that I mean just the fcrepo camel toolbox indexer, not the one for
Drupal (not that it would hurt either). Just avoid alpaca and sync.

**Comment by whikloj** [comment]
`vagrant up`

**Comment by whikloj** [comment]
Not sure that this is problems, just storing these incase trouble crops up.
```
==> default: search was successfully uninstalled.                                        [ok]
==> default: Drupal\Component\Plugin\Exception\PluginNotFoundException: The "search_page" entity type does not exist. in /var/www/html/drupal/core/lib/Drupal/Core/Entity/EntityTypeManager.php:125
==> default: Stack trace:
==> default: #0 /var/www/html/drupal/core/lib/Drupal/Core/Entity/EntityTypeManager.php(225): Drupal\Core\Entity\EntityTypeManager->getDefinition('search_page')
==> default: #1 /var/www/html/drupal/core/lib/Drupal/Core/Entity/EntityTypeManager.php(161): Drupal\Core\Entity\EntityTypeManager->getHandler('search_page', 'storage')
==> default: #2 /var/www/html/drupal/modules/hook_post_action/hook_post_action.module(48): Drupal\Core\Entity\EntityTypeManager->getStorage('search_page')
==> default: #3 /var/www/html/drupal/core/includes/bootstrap.inc(979): _hook_post_action_post_save(Object(Drupal\search\Entity\SearchPage), 'delete')
==> default: #4 [internal function]: _drupal_shutdown_function()
==> default: #5 {main}
```

**Comment by whikloj** [comment]
```
==> default: Installing Karaf as a service... 
==> default: 5039 [sshd-SshClient[7b1d7fff]-nio2-thread-1] WARN org.apache.sshd.client.keyverifier.AcceptAllServerKeyVerifier - Server at /0.0.0.0:8101 presented unverified RSA key: 42:09:f5:cb:1a:f5:f0:5d:a9:c3:99:28:f4:2c:4b:89
==> default: /tmp/vagrant-shell: line 46: Creating: command not found
==> default: insserv: warning: script 'karaf-service' missing LSB tags and overrides
```

**Comment by DiegoPino** [comment]
@whikloj why is "search" being uninstalled?
`==> default: search was successfully uninstalled.                                        [ok]`

**Comment by whikloj** [comment]
@DiegoPino not sure right after the line 
`==> default: #5 {main}`

it says

```
==> default: The following extensions will be enabled: search_api
==> default: Do you really want to continue? (y/n): 
==> default: y
==> default: search_api was enabled successfully.                                        [ok]
==> default: search_api defines the following permissions: administer search_api
```

So perhaps they are being installed in the wrong order?

**Comment by whikloj** [comment]
@ruebot we seem to have lost Loris (#326)

**Comment by whikloj** [comment]
@ruebot we need some changes to our `config.sh` to account for fcrepo-camel-toolbox configuration changes. The URLs now include the scheme, so we need the `http://` to them.

Change this
```
sed -i 's|triplestore.baseUrl=localhost:8080/fuseki/test/update|triplestore.baseUrl=localhost:8080/bigdata/sparql|' "$KARAF_DIR/etc/org.fcrepo.camel.indexing.triplestore.cfg"
```

to 

```
sed -i 's|triplestore.baseUrl=http://localhost:8080/fuseki/test/update|triplestore.baseUrl=http://localhost:8080/bigdata/sparql|' "$KARAF_DIR/etc/org.fcrepo.camel.indexing.triplestore.cfg"
```

**Comment by whikloj** [comment]
@ruebot Doh! 
facepalm
I forgot it was on port 8383

**Comment by whikloj** [comment]
@ruebot ok, so the fcrepo-camel-toolbox config change might be all that we need.

**Comment by DiegoPino** [comment]
@ruebot 👍 thanks for the clarification on the search module uninstall. Errors look safe to be ignored, mostly problems in the uninstall logic that assumes a specific entity should be there. I will do some research and if needed will open a new issue to solve that.

**Comment by MarcusBarnes** [comment]
@ruebot Thanks - I'll have a chance to test in about an hour.

**Comment by whikloj** [comment]
@ruebot can I ask you start new pulls. I was just about to merge and now I must restart testing.

**Comment by dannylamb** [comment]
@ruebot me too

**Comment by MarcusBarnes** [comment]
@ruebot I've tested https://github.com/ruebot/CLAW/tree/issue-357 and it appears to fix issue #374.  Thanks!

**Pull request merged**